core/src/components/action-sheet/action-sheet.scss

```diff
@@ -67,7 +67,7 @@
 
 .action-sheet-wrapper {
   @include position(null, 0, 0, 0);
-  @include margin(var(--ion-safe-area-top, 0), auto, var(--ion-safe-area-bottom, 0), auto);
+  @include margin(var(--ion-safe-area-top, 0), auto, 0, auto);
```

Comment:
This will break the safe area for iOS. The action sheet on iOS should move above the safe area, while the MD action sheet's content should move above the safe area, but the content should extend below the safe area.

I think what was wrong in my previous patch was that iOS safe area should use margin, but MD safe area should use padding.

It might make sense to revert https://github.com/ionic-team/ionic-framework/pull/24176/files and then add 

```
.action-sheet-wrapper {
  @include padding(var(--ion-safe-area-top, 0), auto, var(--ion-safe-area-bottom, 0), auto);
}
```

Or something to the safe are in MD mode

Reply:
Good catch 👍  Although the padding/margin cannot be applied to the `action-sheet-wrapper` in MD, as it will increase the container size (good), but won't increase the container space the background color is applied to (bad). 

Likely we will have to do specific styles in iOS vs. MD stylesheets. As iOS pushes the space from the bottom, but doesn't have a background applied.